book/src/adding_languages.md

```diff
@@ -0,0 +1,51 @@
+# Adding languages
+
+## Submodules
+
+To add a new langauge, you should first add a tree-sitter submodule. To do this, you can run the command
+```sh
+$ git submodule add -f <repository> helix-syntax/languages/tree-sitter-<name>
+```
+For example, to add tree-sitter-ocaml you would run
+```sh
+$ git submodule add -f https://github.com/tree-sitter/tree-sitter-ocaml helix-syntax/languages/tree-sitter-ocaml
+```
+Make sure the submodule is shallow by doing
+```sh
+git config -f .gitmodules submodule.helix-syntax/languages/tree-sitter-<name>.shallow true
+```
+You can also edit the `.gitmodules` file.
+
+## languages.toml
+
+Next, you need to add the language to `languages.toml` found in the root of the repository. The `languages.toml` file is included at compilation time.
+
+These are the available keys and descriptions for the file.
+
+| Key           | Description                                                   |
+| ----          | -----------                                                   |
+| name          | The name of the language                                      |
+| scope         | I don't know what this does                                   |
+| injection-regex | The tree-sitter regex to use for injection. [link](https://tree-sitter.github.io/tree-sitter/syntax-highlighting) |
```

Comment:
I'd link to https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection

I also find the tree-sitter description a bit clearer:
> regex pattern that will be tested against a language name in order to determine whether this language should be used for a potential language injection site.